gen/installer/aws.py

```diff
@@ -219,13 +218,17 @@ def gen_supporting_template():
         })
 
 
-extra_templates = ['aws/dcos-config.yaml', 'coreos-aws/cloud-config.yaml', 'coreos/cloud-config.yaml']
-
-
 def make_advanced_bunch(variant_args, template_name, cc_params):
+    extra_templates = ['aws/dcos-config.yaml', 'aws/templates/advanced/{}'.format(template_name)]
+    if cc_params['os_type'] == 'coreos':
+        extra_templates += ['coreos-aws/cloud-config.yaml', 'coreos/cloud-config.yaml']
+        cloud_init_implementation = 'coreos'
+    else:
+        cloud_init_implementation = 'canonical'
```

Comment:
Yeah, I had a thought about that as well... this tells me we should do it. Adjustments made.